RetailCoder.VBE/Inspections/Concrete/Inspector.cs

```diff
@@ -93,8 +95,13 @@ private void UpdateInspectionSeverity(Configuration config)
                 {
                     LogManager.GetCurrentClassLogger().Error(e);
                 }
+                var issuesByType = allIssues.GroupBy(i => i.GetType())
+                    .ToDictionary(g => g.Key, g => g.ToList());
                 state.OnStatusMessageUpdate(RubberduckUI.ResourceManager.GetString("ParserState_" + state.Status, UI.Settings.Settings.Culture)); // should be "Ready"
-                return allIssues;
+                return issuesByType.Where(kv => kv.Value.Count <= AGGREGATION_THRESHOLD)
+                    .SelectMany(kv => kv.Value)
+                    .Union(issuesByType.Where(kv => kv.Value.Count > AGGREGATION_THRESHOLD)
+                    .Select(kv => new AggregateInspectionResult(kv.Value)));
```

Comment:
This should be solvable in a single pass over issuesByType, but it's still O(n) so it might be acceptable